inc/Addon/WebP/Subscriber.php

```diff
@@ -0,0 +1,570 @@
+
+		$this->options_data = $options_data;
+		$this->options_api  = $options_api;
+
+		if ( ! isset( $server ) && ! empty( $_SERVER ) && is_array( $_SERVER ) ) {
+			$server = $_SERVER;
+		}
+
+		$this->server = $server && is_array( $server ) ? $server : [];
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public static function get_subscribed_events() {
+		return [
+			'rocket_buffer'                     => [ 'convert_to_webp', 16 ],
+			'rocket_disable_webp_cache'         => 'maybe_disable_webp_cache',
+			'rocket_third_party_webp_change'    => 'sync_webp_cache_with_third_party_plugins',
+			'rocket_preload_before_preload_url' => 'add_accept_header',
+		];
+	}
+
+	/**
+	 * Converts images extension to WebP if the file exists.
+	 *
+	 * @since 3.4
+	 *
+	 * @param string $html HTML content.
+	 * @return string
+	 */
+	public function convert_to_webp( $html ) {
+		if ( ! $this->options_data->get( 'cache_webp', 0 ) ) {
+			return $html;
+		}
+
+		/** This filter is documented in inc/classes/buffer/class-cache.php */
+		if ( apply_filters( 'rocket_disable_webp_cache', false ) ) {
+			return $html;
+		}
+
+		if ( ! $this->is_browser_webp_compatible() ) {
+			return $html;
+		}
+
+		$extensions      = $this->get_extensions();
+		$attribute_names = $this->get_attribute_names();
+
+		if (
+			empty( $extensions )
+			||
+			empty( $attribute_names )
+		) {
+			return $html . '<!-- Rocket no webp -->';
+		}
+
+		$extensions      = implode( '|', $extensions );
+		$attribute_names = implode( '|', $attribute_names );
+
+		if ( ! preg_match_all( '@["\'\s](?<name>(?:data-(?:[a-z0-9_-]+-)?)?(?:' . $attribute_names . '))\s*=\s*["\']\s*(?<value>(?:https?:/)?/[^"\']+\.(?:' . $extensions . ')[^"\']*?)\s*["\']@is', $html, $attributes, PREG_SET_ORDER ) ) {
+			return $html . '<!-- Rocket no webp -->';
+		}
+
+		if ( ! isset( $this->filesystem ) ) {
+			$this->filesystem = \rocket_direct_filesystem();
+		}
+
+		$has_webp = false;
+
+		foreach ( $attributes as $attribute ) {
+			if ( preg_match( '@srcset$@i', strtolower( $attribute['name'] ) ) ) {
+				// This is a srcset attribute, with probably multiple URLs.
+				$new_value = $this->srcset_to_webp( $attribute['value'], $extensions );
+			} else {
+				// A single URL attibute.
+				$new_value = $this->url_to_webp( $attribute['value'], $extensions );
+			}
+
+			if ( ! $new_value ) {
+				// No webp here.
+				continue;
+			}
+
+			// Replace in content.
+			$has_webp = true;
+			$new_attr = preg_replace( '@' . $attribute['name'] . '\s*=\s*["\'][^"\']+["\']@s', $attribute['name'] . '="' . $new_value . '"', $attribute[0] );
+			$html     = str_replace( $attribute[0], $new_attr, $html );
+		}
+
+		/**
+		 * Tell if the page contains webp files.
+		 *
+		 * @since 3.4
+		 *
+		 * @param bool   $has_webp True if the page contains webp files. False otherwise.
+		 * @param string $html     The page’s html contents.
+		 */
+		$has_webp = apply_filters( 'rocket_page_has_hebp_files', $has_webp, $html );
```

Comment:
I know that this filter was there before this refactoring but I believe there is a typo on the filter name and u solved it for variable name, I'm not sure we can change that and deprecate the old name, what do u think?
```suggestion
		$has_webp = apply_filters( 'rocket_page_has_webp_files', $has_webp, $html );
```